app/assets/javascripts/form-validator.js

```diff
@@ -37,15 +37,23 @@ var FormValidator = function(root) {
 
   this.addInputValidation = function(inputId, validator) {
     var onChange = function (input) {
+      var inputValidationError = root.getElementById(input.id + 'ValidationError');
       if (validator.isValid(input.value)) {
+        inputValidationError.classList.add('display-none');
         input.classList.add('valid');
       } else {
+        if(input.value.length>0){
+          inputValidationError.classList.remove('display-none');
+        }
         input.classList.remove('valid');
+
       }
     };
     var input = document.getElementById(inputId);
     input.addEventListener('change', onChange.bind(this, input));
```

Comment:
Clarity is arguable and always subjective, personally I find using target more idiomatic, for example.
Also, there's absolutely no trouble with using event target for this simple case, as only ancient browsers (early versions of Firefox and IE8 and below) do not comply with the standard.

Primary concern here is to avoid this style as a habit, because in a bigger application this approach may cause memory leaks as it implicitly prevents the DOM node it stores a reference to from being collected.

And even if it's still the way you'd like to do it, closures would be a little better, I guess.